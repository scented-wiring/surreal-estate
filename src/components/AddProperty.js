import React, { useState } from 'react';
import "../styles/AddProperty.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: '',
      city: 'Manchester',
      type: 'Flat',
      bedrooms: '',
      bathroomss: '',
      price: '',
      email: ''
    }
  }

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    console.log(fields);
  }

  const handleFieldChange = (event) => {
    setFields({...fields, [event.target.name]: event.target.value });
  }

  return (
    <div className="AddProperty">
      <form onSubmit={handleAddProperty}>
        <label htmlFor='title'>
          Title
          <input id="title" name="title" value={fields.title} onChange={handleFieldChange} placeholder="2 bed flat" className="input-title" />
        </label>
        <label htmlFor="city">
          City
          <select id="city" name="city" value={fields.city} onChange={handleFieldChange} className="input-city">
            <option value="Manchester">Manchester</option>
            <option value="Leeds">Leeds</option>
            <option value="Sheffield">Sheffield</option>
            <option value="Liverpool">Liverpool</option>
          </select>
        </label>
        <label htmlFor="type">
          Type
          <select id="type" name="type" value={fields.type} onChange={handleFieldChange} className="select-type">
            <option value="Flat">Flat</option>
            <option value="Detached">Detached</option>
            <option value="Semi-Detached">Semi-Detached</option>
            <option value="End of Terrace">End of Terrace</option>
            <option value="Cottage">Cottage</option>
            <option value="Bungalow">Bungalow</option>
          </select>
        </label>
        <label htmlFor="bedrooms">
          Bedrooms
          <select id="bedrooms" name="bedrooms" value={fields.bedrooms} onChange={handleFieldChange} className="select-bedrooms">
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
          </select>
        </label>
        <label htmlFor="bathrooms">
          Bathrooms
          <select id="bathrooms" name="bathrooms" value={fields.bathrooms} onChange={handleFieldChange} type="number" className="select-bathrooms" >
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
          </select>
        </label>
        <label htmlFor="price">
          Price Â£
          <input id="price" name="price" value={fields.price} onChange={handleFieldChange} type="number" className="input-price" placeholder="25000" min="1" max="999999" />
        </label>
        <label htmlFor="email">
          Email
          <input id="email" name="email" value={fields.email} onChange={handleFieldChange} type="email" placeholder="example@email.com" className="input-email" />
        </label>
        <button type="submit">Add</button>
      </form>
    </div>
  )
}

export default AddProperty;