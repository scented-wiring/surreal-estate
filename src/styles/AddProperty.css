.AddProperty {
  margin-left: auto;
  margin-right: auto;
  margin-top: 8px;
  grid-column: 1 / 4;
  grid-row: 3 / 4;
  display: grid;
  background-color: lightgray;
  padding: 15px;
  width: 50%;
}

.form-add-property {
  display: flex;
  flex-direction: column;
  padding: 10px;
}

label {
  display: grid;
  margin-top: 10px;
  margin-left: 10px;
  margin-right: 10px;
}

.add-button {
  margin-top: 10px;
  margin-left: 10px;
  margin-right: 10px;
  width: 60px;
  color: #fff !important;
  background: purple;
  padding: 3px;
  border-radius: 5px;
  display: inline-block;
  border: none;
  transition: all 0.4s ease 0s;
}

.add-button:hover {
  background: #363636;
  transition: all 0.4s ease 0s;
}
