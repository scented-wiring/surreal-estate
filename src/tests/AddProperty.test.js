import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  it("renders a component", () => {
    const { asFragment } = render(<AddProperty />);
    expect(asFragment).toMatchSnapshot();
  });

  it("renders an AddProperty form", () => {
    const { getByTestId } = render(<AddProperty />);
    expect(getByTestId("add-property-form")).toBeInTheDocument();
  });

  it("renders 2 textboxes", () => {
    const { getAllByRole } = render(<AddProperty />);
    expect(getAllByRole("textbox")).toHaveLength(2);
  });

  it("renders 4 comboboxes", () => {
    const { getAllByRole } = render(<AddProperty />);
    expect(getAllByRole("combobox")).toHaveLength(4);
  });

  it("renders 1 spinbutton", () => {
    const { getAllByRole } = render(<AddProperty />);
    expect(getAllByRole("spinbutton")).toHaveLength(1);
  });

  it("renders 1 button", () => {
    const { getAllByRole } = render(<AddProperty />);
    expect(getAllByRole("button")).toHaveLength(1);
  });
});
