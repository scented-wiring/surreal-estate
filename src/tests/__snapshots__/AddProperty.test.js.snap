// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders a component 1`] = `[Function]`;
